src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -174,123 +166,145 @@ Path ThetaStarPathPlanner::findPath(const Point &start, const Point &destination
                                     const Rectangle &navigable_area,
                                     const std::vector<Obstacle> &obstacles)
 {
-    obstacles_ = obstacles;
-    Path empty_ret_val(std::nullopt);
-    CellCoordinate src_coord, dest_coord;
-
-    openList.clear();
+    // Initialize member variables
+    this->obstacles       = obstacles;
+    max_navigable_x_coord = navigable_area.xLength() / 2.0 - ROBOT_MAX_RADIUS_METERS;
+    max_navigable_y_coord = navigable_area.yLength() / 2.0 - ROBOT_MAX_RADIUS_METERS;
+    num_grid_rows = (int)((max_navigable_x_coord * 2.0 + ROBOT_MAX_RADIUS_METERS) /
+                          SIZE_OF_GRID_CELL_IN_METERS);
+    num_grid_cols = (int)((max_navigable_y_coord * 2.0 + ROBOT_MAX_RADIUS_METERS) /
+                          SIZE_OF_GRID_CELL_IN_METERS);
+
+    // Reset data structures to path plan again
+    open_list.clear();
+    closed_list.clear();
     unblocked_grid.clear();
```

Comment:
Could we put all the member initialization in it's own function (so the "Initialize member variables" and "Reset data ..." sections here)?

Reply:
done